http/src/lib.rs

```diff
@@ -562,45 +579,67 @@ fn configure_port(_reuse: bool, _tcp: &net2::TcpBuilder) -> io::Result<()> {
 	Ok(())
 }
 
+/// Handle used to close the server. Can be cloned and passed around to different threads and be used
+/// to close a server that is `wait()`ing.
+#[derive(Clone)]
+pub struct CloseHandle {
+	executors: Arc<Mutex<Option<Vec<Executor>>>>,
+	closers: Arc<Mutex<Option<Vec<oneshot::Sender<()>>>>>,
+}
+
+impl CloseHandle {
+	/// Shutdown a running server
+	pub fn close(self) {
+        if let Some(executors) = self.executors.lock().take() {
```

Comment:
Since we always lock both, might be worth introducing a struct for that.

Reply:
…like you suggested on the other PR! Indeed, you're right.

[This commit](https://github.com/paritytech/jsonrpc/pull/437/commits/b271b119bedf3bf215d381657404d0d9c55ab9dd) uses:

```
#[derive(Clone)]
pub struct CloseHandle(Arc<Mutex<Option<Vec<(Executor, oneshot::Sender<()>)>>>>);
```